controllers/api/v1/search_test.go

```diff
@@ -0,0 +1,318 @@
+
+						Expect(response.StatusCode).To(Equal(http.StatusUnprocessableEntity))
+					})
+
+					It("returns correct json response", func() {
+						user := FabricateUser("John", "john@example.com", "secret")
+						body := CreateEmptyMultipartBody()
+						client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+						oauthToken := FabricateOAuthToken(client, user.Id)
+						expectedResponse := `{
+							"errors": [
+								{
+									"detail": "File can't be blank."
+								}
+							]
+						}`
+
+						header := http.Header{
+							"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())},
+							"Content-Type":  {"multipart/form-data; boundary=multipart-boundary"}}
+
+						response := MakeAuthenticatedRequest("POST", "/api/v1/search", header, body, nil)
+						responseBody, err := ioutil.ReadAll(response.Body)
+						if err != nil {
+							Fail("Reading response body failed: " + err.Error())
+						}
+
+						Expect(string(responseBody)).To(MatchJSON(expectedResponse))
+					})
+				})
+
+				Context("given the CSV file is wrong formatted", func() {
+					It("returns 401 status code", func() {
+						user := FabricateUser("John", "john@example.com", "secret")
+						filePath := AppRootDir() + "/tests/fixtures/shared/invalid_keyword.csv"
+						_, body := CreateMultipartFormData(filePath)
+						client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+						oauthToken := FabricateOAuthToken(client, user.Id)
+
+						header := http.Header{
+							"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())},
+							"Content-Type":  {"multipart/form-data; boundary=multipart-boundary"}}
+
+						response := MakeAuthenticatedRequest("POST", "/api/v1/search", header, body, nil)
+
+						Expect(response.StatusCode).To(Equal(http.StatusUnprocessableEntity))
+					})
+
+					It("returns correct json response", func() {
+						user := FabricateUser("John", "john@example.com", "secret")
+						filePath := AppRootDir() + "/tests/fixtures/shared/invalid_keyword.csv"
+						_, body := CreateMultipartFormData(filePath)
+						client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+						oauthToken := FabricateOAuthToken(client, user.Id)
+						expectedResponse := `{
+							"errors": [
+								{
+									"detail": "CSV contents are not in correct format."
+								}
+							]
+						}`
+
+						header := http.Header{
+							"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())},
+							"Content-Type":  {"multipart/form-data; boundary=multipart-boundary"}}
+
+						response := MakeAuthenticatedRequest("POST", "/api/v1/search", header, body, nil)
+						responseBody, err := ioutil.ReadAll(response.Body)
+						if err != nil {
+							Fail("Reading response body failed: " + err.Error())
+						}
+
+						Expect(string(responseBody)).To(MatchJSON(expectedResponse))
+					})
+				})
+
+				Context("given the file is an image", func() {
```

Comment:
How about making the test description more generic:

```suggestion
				Context("given an invalid file type", func() {
```